src/langgraph_google_alloydb_pg/engine.py

```diff
@@ -0,0 +1,530 @@
+        cls,
+        url: Union[str | URL],
+        **kwargs: Any,
+    ) -> AlloyDBEngine:
+        """Create an AlloyDBEngine instance from arguments
+
+        Args:
+            url (Optional[str]): the URL used to connect to a database. Use url or set other arguments.
+
+        Raises:
+            ValueError: If not all database url arguments are specified
+
+        Returns:
+            AlloyDBEngine
+        """
+        # Running a loop in a background thread allows us to support
+        # async methods from non-async environments
+        if cls._default_loop is None:
+            cls._default_loop = asyncio.new_event_loop()
+            cls._default_thread = Thread(
+                target=cls._default_loop.run_forever, daemon=True
+            )
+            cls._default_thread.start()
+
+        driver = "postgresql+asyncpg"
+        if (isinstance(url, str) and not url.startswith(driver)) or (
+            isinstance(url, URL) and url.drivername != driver
+        ):
+            raise ValueError("Driver must be type 'postgresql+asyncpg'")
+
+        engine = create_async_engine(url, **kwargs)
+        return cls(cls.__create_key, engine, cls._default_loop, cls._default_thread)
+
+    async def _run_as_async(self, coro: Awaitable[T]) -> T:
+        """Run an async coroutine asynchronously"""
+        # If a loop has not been provided, attempt to run in current thread
+        if not self._loop:
+            return await coro
+        # Otherwise, run in the background thread
+        return await asyncio.wrap_future(
+            asyncio.run_coroutine_threadsafe(coro, self._loop)
+        )
+
+    def _run_as_sync(self, coro: Awaitable[T]) -> T:
+        """Run an async coroutine synchronously"""
+        if not self._loop:
+            raise Exception(
+                "Engine was initialized without a background loop and cannot call sync methods."
+            )
+        return asyncio.run_coroutine_threadsafe(coro, self._loop).result()
+    
+    async def close(self) -> None:
+        """Dispose of connection pool"""
+        await self._pool.dispose()
+
+    async def _ainit_checkpoint_table(
+        self, schema_name: str = "public"
+    ) -> None:
+        """
+        Create AlloyDB tables to save checkpoints.
+
+        Args:
+            schema_name (str): The schema name to store the checkpoint tables.
+                Default: "public".
+
+        Returns:
+            None
+        """
+        create_checkpoints_table = f"""CREATE TABLE IF NOT EXISTS "{schema_name}".checkpoints(
+            thread_id TEXT NOT NULL,
+            checkpoint_ns TEXT NOT NULL DEFAULT '',
+            checkpoint_id TEXT NOT NULL,
+            parent_checkpoint_id TEXT,
+            v INTEGER NOT NULL,
+            checkpoint JSONB NOT NULL,
+            metadata JSONB NOT NULL DEFAULT '{{}}',
+            channel TEXT NOT NULL,
+            version TEXT NOT NULL,
+            type TEXT,
+            blob BYTEA,
```

Comment:
Remove type and blob. In the schema we discussed expanding the checkpoint and metadata objects into the table. This doesn't exactly match the doc or the warning check or the puts SQL statement.  We can see how just inserting the checkpoint and metadata objects as JSON into the table and see how that works.